core/drivers/crypto/caam/cipher/caam_cipher.c

```diff
@@ -0,0 +1,1052 @@
+
+	/* Check if all required keys are defined */
+	if (alg->require_key & NEED_KEY1) {
+		if (!dinit->key1.data || !dinit->key1.length)
+			goto exit_init;
+
+		retstatus = do_check_keysize(&alg->def_key, dinit->key1.length);
+		if (retstatus != CAAM_NO_ERROR) {
+			CIPHER_TRACE("Bad Key 1 size");
+			goto exit_init;
+		}
+
+		/* Copy the key 1 */
+		retstatus = copy_ctx_data(&cipherdata->key1, &dinit->key1);
+		CIPHER_TRACE("Copy Key 1 returned 0x%" PRIx32, retstatus);
+
+		if (retstatus != CAAM_NO_ERROR) {
+			ret = TEE_ERROR_OUT_OF_MEMORY;
+			goto exit_init;
+		}
+	}
+
+	if (alg->require_key & NEED_KEY2) {
+		if (!dinit->key2.data || !dinit->key2.length)
+			goto exit_init;
+
+		retstatus = do_check_keysize(&alg->def_key, dinit->key2.length);
+		if (retstatus != CAAM_NO_ERROR) {
+			CIPHER_TRACE("Bad Key 2 size");
+			goto exit_init;
+		}
+
+		/* Copy the key 2 */
+		retstatus = copy_ctx_data(&cipherdata->key2, &dinit->key2);
+		CIPHER_TRACE("Copy Key 2 returned 0x%" PRIx32, retstatus);
+
+		if (retstatus != CAAM_NO_ERROR) {
+			ret = TEE_ERROR_OUT_OF_MEMORY;
+			goto exit_init;
+		}
+	}
+
+	if (alg->require_key & NEED_IV) {
+		if (!dinit->iv.data || !dinit->iv.length)
+			goto exit_init;
+
+		if (dinit->iv.length != alg->size_ctx) {
+			CIPHER_TRACE("Bad IV size %zu (expected %" PRId32 ")",
+				     dinit->iv.length, alg->size_ctx);
+			goto exit_init;
+		}
+
+		CIPHER_TRACE("Allocate CAAM Context Register (%" PRId32
+			     " bytes)",
+			     alg->size_ctx);
+
+		/* Copy the IV into the context register */
+		retstatus = copy_ctx_data(&cipherdata->ctx, &dinit->iv);
+		CIPHER_TRACE("Copy IV returned 0x%" PRIx32, retstatus);
+
+		if (retstatus != CAAM_NO_ERROR) {
+			ret = TEE_ERROR_OUT_OF_MEMORY;
+			goto exit_init;
+		}
+	}
+
+	if (alg->require_key & NEED_TWEAK) {
+		/* This is accepted to start with a NULL Tweak */
+		if (dinit->iv.length) {
+			if (dinit->iv.length != alg->size_block) {
+				CIPHER_TRACE("Bad tweak 2 size");
+				goto exit_init;
+			}
+
+			/* Copy the tweak */
+			retstatus = copy_ctx_data(&cipherdata->tweak,
+						  &dinit->iv);
+			CIPHER_TRACE("Copy Tweak returned 0x%" PRIx32,
+				     retstatus);
+
+			if (retstatus != CAAM_NO_ERROR) {
+				ret = TEE_ERROR_OUT_OF_MEMORY;
+				goto exit_init;
+			}
+		} else {
+			/* Create tweak 0's */
+			if (!cipherdata->tweak.data) {
+				/*
+				 * Allocate the destination buffer and
+				 * fill it with 0's
+				 */
+				ret = caam_calloc_align_buf(&cipherdata->tweak,
+							    alg->size_block);
+				if (ret != CAAM_NO_ERROR)
+					return ret;
+			} else {
+				/* Fill it with 0's */
+				memset(cipherdata->tweak.data, 0,
+				       cipherdata->tweak.length);
+
+				/* Push data to physical memory */
+				cache_operation(TEE_CACHEFLUSH,
+						cipherdata->tweak.data,
+						cipherdata->tweak.length);
+			}
+		}
+	}
+
+	/* Save the operation direction */
+	cipherdata->encrypt = dinit->encrypt;
+	cipherdata->blockbuf.filled = 0;
+
+	ret = TEE_SUCCESS;
+
+exit_init:
```

Comment:
We'd normally name this label `out:`, there's no need to involve the name of the function. I've seen this in a few functions in at least this file in this PR.

Reply:
Will rename labels `out:` instead for functions having one label